logstash-core/src/main/java/org/logstash/config/ir/compiler/EventCondition.java

```diff
@@ -0,0 +1,709 @@
+
+            private FieldInField(final FieldReference left, final FieldReference right) {
+                this.left = left;
+                this.right = right;
+            }
+
+            @Override
+            public boolean fulfilled(final JrubyEventExtLibrary.RubyEvent event) {
+                final Object lfound = event.getEvent().getUnconvertedField(left);
+                final Object rfound = event.getEvent().getUnconvertedField(right);
+                if (lfound instanceof ConvertedList || lfound instanceof ConvertedMap) {
+                    return false;
+                } else if (lfound instanceof RubyString && rfound instanceof RubyString) {
+                    return ((RubyString) lfound).getByteList()
+                        .indexOf(((RubyString) rfound).getByteList()) > -1;
+                } else if (rfound instanceof ConvertedList) {
+                    return contains((ConvertedList) rfound, lfound);
+                } else {
+                    return lfound != null && rfound != null && lfound.equals(rfound);
+                }
+            }
+        }
+
+        private static final class FieldInConstantList implements EventCondition {
+
+            private final FieldReference field;
+
+            private final List<?> value;
+
+            private FieldInConstantList(final FieldReference field, final List<?> value) {
+                this.field = field;
+                this.value = value;
+            }
+
+            @Override
+            public boolean fulfilled(final JrubyEventExtLibrary.RubyEvent event) {
+                final Object found = event.getEvent().getUnconvertedField(field);
+                return found != null &&
+                    value.stream().anyMatch(val -> val.toString().equals(found.toString()));
+            }
+        }
+
+        private static final class FieldTruthy implements EventCondition {
+
+            private final FieldReference field;
+
+            private FieldTruthy(final FieldReference field) {
+                this.field = field;
+            }
+
+            @Override
+            public boolean fulfilled(final JrubyEventExtLibrary.RubyEvent event) {
+                final Object object = event.getEvent().getUnconvertedField(field);
+                if (object == null) {
+                    return false;
+                }
+                final String other = object.toString();
+                return other != null && !other.isEmpty() &&
+                    !Boolean.toString(false).equals(other);
+            }
+        }
+
+        /**
+         * This {@link IllegalArgumentException} is thrown when the inputs to an {@code if}
+         * condition do not conform to the expected types. It being thrown is a bug in Logstash
+         * in every case.
+         */
+        private static final class UnexpectedTypeException extends IllegalArgumentException {
+
+            private static final long serialVersionUID = -3673305675726770832L;
```

Comment:
What's with the super-long number? Why not, say, `1L`?

Not a problem, just curious.

Reply:
@andrewvc Because I clicked auto generate in idea :D set to `1L` :)